src/loaddb/load_db.c

```diff
@@ -1463,7 +1469,48 @@ ldr_check_file_list (std::string & file_name, int &num_files, int &error_code)
       num_files++;
 
       strcpy (schema_object_file->schema_file_name, buffer);
-      read_file_name = buffer;
+
+      last_backslash = schema_info_fullpath.find_last_of (PATH_SEPARATOR);
+
+      if (last_backslash == std::string::npos)
+	{
+	  last_backslash = 0;
+	}
+
+#if defined(WINDOWS)
+      last_slash = schema_info_fullpath.find_last_of ('/');
+
+      if (last_slash == std::string::npos)
+	{
+	  last_slash = 0;
+	}
+
+      if (last_slash > last_backslash)
+	{
+	  schema_info_fullpath = schema_info_fullpath.substr (0, last_slash + 1);
+	  read_file_name = schema_info_fullpath + buffer;
+	}
+      else if (last_slash < last_backslash)
+	{
+	  schema_info_fullpath = schema_info_fullpath.substr (0, last_backslash + 1);
+	  read_file_name = schema_info_fullpath + buffer;
+	}
+      else
+	{
+	  read_file_name = buffer;
+	}
+#else
+      if (last_backslash > 0)
+	{
+	  schema_info_fullpath = schema_info_fullpath.substr (0, last_backslash + 1);
+	  read_file_name = schema_info_fullpath + buffer;
+	}
+      else
+	{
+	  read_file_name = buffer;
+	}
+#endif
+
```

Comment:
Maybe modified code itself could working without problem. I propose minor change, just refer and do it as you like.
1. Change variable 'last_backslash' to 'seperator_pos'
2. change modified line#: 1472 ~ 1513 to the code below:
```
      seperator_pos = schema_info_fullpath.find_last_of (PATH_SEPARATOR);

      if (seperator_pos == std::string::npos)
        {
          seperator_pos = 0;
        }

#if defined(WINDOWS)
      last_slash = schema_info_fullpath.find_last_of ('/');

      if (last_slash == std::string::npos)
        {
          last_slash = 0;
        }

      if (last_slash > seperator_pos)
        {
          seperator_pos = last_slash;
        }
#endif

      if (seperator_pos > 0)
        {
          schema_info_fullpath = schema_info_fullpath.substr (0, seperator_pos + 1);
          read_file_name = schema_info_fullpath + buffer;
        }
      else
        {
          read_file_name = buffer;
        }

```

Reply:
I will modify it as per your suggestion.